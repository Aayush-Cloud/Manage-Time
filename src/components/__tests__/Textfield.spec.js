// @flow
import React from 'react';
import { TextInput } from 'react-native';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Textfield from '../Textfield.ios';

const props = {
  input: {
    onChange: jest.fn(),
    value: 'some value',
  },
  value: 'another value',
};

describe('Textfield component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Textfield {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders error messages correctly', () => {
    const otherProps = {
      ...props,
      meta: {
        error: 'an error message',
        submitFailed: true,
        touched: true,
      },
    };
    const tree = renderer.create(<Textfield {...otherProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('uses props.input.value as value if defined', () => {
    const wrapper = shallow(<Textfield {...props} />);
    const textfield = wrapper.find(TextInput);
    expect(textfield.prop('value')).toBe(props.input.value);
  });

  it('uses props.value as value if props.input.value is not defined', () => {
    const otherProps = {
      input: {
        onChange: () => { },
      },
      value: 'another value',
    };
    const wrapper = shallow(<Textfield {...otherProps} />);
    const textfield = wrapper.find(TextInput);
    expect(textfield.prop('value')).toBe(otherProps.value);
  });

  it('calls input.onChange with new input when input changes', () => {
    const wrapper = shallow(<Textfield {...props} />);
    const textfield = wrapper.find(TextInput);
    textfield.simulate('changeText', 'new text');
    const { onChange } = props.input;
    expect(onChange.mock.calls.length).toBe(1);
    expect(onChange.mock.calls[0][0]).toBe('new text');
  });
});
