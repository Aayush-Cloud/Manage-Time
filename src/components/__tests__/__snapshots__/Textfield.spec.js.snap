// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textfield component renders correctly 1`] = `
<View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <TextInput
        clearButtonMode="while-editing"
        editable={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholderTextColor="rgba(0, 0, 0, 0.38)"
        style={
          Object {
            "alignItems": "flex-end",
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "Helvetica Neue",
            "fontSize": 16,
            "fontWeight": "400",
            "height": 28,
            "lineHeight": 16,
          }
        }
        value="some value"
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "height": 1,
        },
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.12)",
        },
      ]
    }
  />
</View>
`;

exports[`Textfield component renders error messages correctly 1`] = `
<View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <TextInput
        clearButtonMode="while-editing"
        editable={true}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholderTextColor="rgba(0, 0, 0, 0.38)"
        style={
          Object {
            "alignItems": "flex-end",
            "color": "rgba(0, 0, 0, 0.87)",
            "fontFamily": "Helvetica Neue",
            "fontSize": 16,
            "fontWeight": "400",
            "height": 28,
            "lineHeight": 16,
          }
        }
        value="some value"
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "height": 1,
        },
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0.12)",
        },
      ]
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#ff1744",
        "fontFamily": "Helvetica Neue",
        "fontSize": 14,
        "marginTop": 8,
      }
    }
  >
    an error message
  </Text>
</View>
`;
